Splitting expenses

jyuw opened the pull request:
#### PT Story: https://www.pivotaltracker.com/story/show/155625606

#### Description
Changes proposed in this pull request:

* adds algorithm for splitting expenses within a group 
* displays who owes whom 

#### What I have learned working on this feature: 
* pry is love, pry is life
* accessing hash values
* render vs redirect_to
* nested loops
* using FactoryBot with associations

#### Screenshots:

![screen shot 2018-03-03 at 12 04 55 pm](https://user-images.githubusercontent.com/34248712/36933570-aa72a02c-1edb-11e8-9026-62e0452c5788.png)

jyuw committed:
added steps for creating table of expenses

jyuw committed:
added extraction of expenses and ids for ease of use
added calculation for who owes whom

jyuw committed:
WIP printing our who owes who

jyuw committed:
added split method where lenders and debtors are separated and their amounts are compared WIP

holgertidemand committed:
WIP output function, stuck with until loop
trying to modify hashes values so they cancel out

holgertidemand committed:
WIP compering lenders and debtors and saves them in output array

jyuw committed:
fixes redirection to group page after creation
WIP printing out output message on show page

tochman commented on app/views/groups/show.html.erb:
```diff
@@ -25,4 +25,6 @@
 
-<h3> WHO OWES WHO!?!!?!</h3>
+<h3> Who owes whom? </h3>
+<% binding.pry %>
```
Please remove this ;-)

tochman commented on features/expenses_can_be_split.feature:
```diff
@@ -0,0 +1,45 @@
+      | jade@ca.com  | cheese      | 1000   | Holgers trip |
+    And I am logged in as "jade@ca.com"
+
+  Scenario:
```
I would like to see a scenario title here.

tochman commented on features/expenses_can_be_split.feature:
```diff
@@ -0,0 +1,45 @@
+    And I should see "Total expenses per person: 1200kr"
+
+
+  Scenario:
```
..and here...

tochman commented on features/step_definitions/expenses_steps.rb:
```diff
@@ -4,11 +4,23 @@
+  table.hashes.each do |group|
+    current_group = Group.find_by(name: group[:group])
+    user = User.find_by(email: group[:email])
+    expense = Expense.create(description: group[:description], amount: group[:amount])
```
Why not use the factory?

jyuw committed:
slight refactoring to expenses controller
able to print output on the view
WIP splitting within the group when one member hasnt paid an expense

jyuw committed:
changed average calculation to account for members who dont contribute
WIP checking in the if statement if lender or debtor is 0

davidboll committed:
Fixes calculations for app
All green

davidboll committed:
removes output for "owes 0"

jyuw committed:
refactored and cleaned up files
made split method more readable
rewrote cucumber test for splitting to make sure expenses were created dynamically
removed mailer files from view

holgertidemand committed:
adds .env to gitignore

jyuw committed:
fixed user stories to be consistent

jyuw committed:
Merge branch 'splitting-expenses' of https://github.com/jyuw/evensteven into splitting-expenses

diraulo commented on app/controllers/expenses_controller.rb:
```diff
@@ -17,4 +20,59 @@ def add_current_user_to_expense
+      end
+    end
+
+    lenders.each do |lender|
```
This block looks less cryptic from what I saw yesterday 🙂

diraulo commented on app/views/groups/show.html.erb:
```diff
@@ -2,25 +2,39 @@
+<% end %>
+
+<% if @output %>
+  <h3>
```
Why not keep this `h3` in a single line?

diraulo commented on app/views/layouts/mailer.html.erb:
```diff
@@ -1,13 +0,0 @@
-<!DOCTYPE html>
```
Any reason why this file was removed? Does it have any effect on this feature? 🤔

holgertidemand replied:
We didnit use it!  got it from devise i think!

diraulo replied:
This file comes from the rails scaffold..

holgertidemand replied:
still works with out it ;)

jyuw replied:
Oops :p should we put it back?

diraulo replied:
I guess it'll take a while before you learn not to touch things that aren't directly related to the feature you're working on huh? 🤔 🙂

Not a blocker really... Just trying to keep you guys in line

tochman replied:
I would put it back. Yeah, just put it back.

diraulo reviewed (commented):
I see you guys managed to break CI? 🤔 

Also, fix the merge conflict on this branch.

jyuw committed:
Merge branch 'develop' into splitting-expenses

jyuw committed:
changed user stories
merged upstream
changed h3 tag in groups/show

jyuw committed:
Merge branch 'splitting-expenses' of https://github.com/jyuw/evensteven into splitting-expenses

tochman committed:
Merge branch 'develop' into splitting-expenses
